### PR TITLE
Avoid implicit interpretation of return type for dyn_aes_encrypt_msg()

### PR DESCRIPTION
**Pull request by smukil**
dyn_aes_encrypt_msg() returns a type of rstatus_t, however the function
returns a status only on an error and returns a 'count' on success
which the caller implicitly interprets as '# of bytes'. This pattern
is error prone and could lead to hard to track down bugs.

This patch also returns the right error code on the OOM case and also
adds checking on the caller side for this new error code DN_ENOMEM.

This was noticed while debugging a core dump. While this patch is only
for clean up, a future patch will change some more code in this area
to fix a crash bug.

### CODE REVIEW

**Comment by shailesh33** [comment]
Thank you. Theres also some notion about error codes. There are some error codes for dynomite and some from redis and some from system. May be unification of these will be helpful?

**Comment by shailesh33** [comment]
Sorry, I will still be poking my nose around this repo :)

**Comment by shailesh33** [comment]
Looks good

**Comment by smukil** [comment]
> Sorry, I will still be poking my nose around this repo :)

Please do! Would much appreciate your input/guidance whenever possible.

I think that we should take actions based on the Redis error codes only in Redis specific functions, which I think mostly live in src/proto/dyn_redis.c.

Otherwise those errors should probably be sent back directly to the client, or we should interpret it to a Dynomite equivalent error code if we want to take actions in other parts of the code that are not Redis specific.

Eg: If Redis returns OOM, then we should reinterpret that as DN_ENOMEM _if_ we want to take some action based on that in parts of the code that are not redis specific. I've not gone deep in the code to know whether this is already happening or not, but that's my general take on it.

So to summarize, I think your suggestion of unification from a general Dynomite point of view will be helpful. So we'll have generic code to deal with failures from memcache or Redis (or any other storage engine that we may support) in the same way, given that they're not engine specific errors.

**Review by ipapapa — APPROVED** [review]
